tests/filecheck/transforms/convert_memref_to_ptr.mlir

```diff
@@ -66,6 +66,27 @@ memref.store %fv, %farr[%idx] {"nontemporal" = false} : memref<10xf64>
 // CHECK-NEXT:  %flv2 = ptr_xdsl.to_ptr %fmem : memref<f64> -> !ptr_xdsl.ptr
 // CHECK-NEXT:  %flv2_1 = ptr_xdsl.load %flv2 : !ptr_xdsl.ptr -> f64
 
+%subview = memref.subview %arr[5][5][1] : memref<10xi32> to memref<5xi32>
+
+// CHECK-NEXT: %3 = arith.constant 5 : index
+// CHECK-NEXT: %bytes_per_element_6 = ptr_xdsl.type_offset i32 : index
+// CHECK-NEXT: %scaled_pointer_offset_6 = arith.muli %3, %bytes_per_element_6 : index
+// CHECK-NEXT: %subview = ptr_xdsl.to_ptr %arr : memref<10xi32> -> !ptr_xdsl.ptr
+// CHECK-NEXT: %offset_pointer_6 = ptr_xdsl.ptradd %subview, %scaled_pointer_offset_6 : (!ptr_xdsl.ptr, index) -> !ptr_xdsl.ptr
+
+%size, %dyn = "test.op"() : () -> (index, memref<?xi32>)
+%dynsubview = memref.subview %dyn[%idx][%size][1] : memref<?xi32> to memref<?xi32>
+
+// CHECK: %bytes_per_element_7 = ptr_xdsl.type_offset i32 : index
+// CHECK-NEXT: %scaled_pointer_offset_7 = arith.muli %idx, %bytes_per_element_7 : index
+// CHECK-NEXT: %dynsubview = ptr_xdsl.to_ptr %dyn : memref<?xi32> -> !ptr_xdsl.ptr
+// CHECK-NEXT: %offset_pointer_7 = ptr_xdsl.ptradd %dynsubview, %scaled_pointer_offset_7 : (!ptr_xdsl.ptr, index) -> !ptr_xdsl.ptr
+
+%reinterpret_cast = memref.reinterpret_cast %arr to offset: [0], sizes: [5, 2], strides: [1, 1] : memref<10xi32> to memref<5x2xi32>
+
+// CHECK-NEXT: %reinterpret_cast = ptr_xdsl.to_ptr %arr : memref<10xi32> -> !ptr_xdsl.ptr
```

Comment:
this is incorrect, the types coming in should be the same as the types going out

Reply:
I don't think I understand. The way I convert this produces
```
%reinterpret_cast = ptr_xdsl.to_ptr %arr : memref<10xi32> -> !ptr_xdsl.ptr
%reinterpret_cast_1 = unrealized_conversion_cast %reinterpret_cast : !ptr_xdsl.ptr -> memref<5x2xi32>
```
and the cast gets erased by the reconcile rewrite since it satisfies its second condition: all uses are ToPtrOps because there are no uses - I can add a condition on uses being greater than zero?

Reply:
I just don't think we want to do reconciling casts like this, I'll fix this in two PRs today.

Reply:
This is now ready to update

Reply:
Ah no, sorry, waiting on this: 4342

Reply:
This is now in, let's update when you have time